src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -187,7 +193,18 @@ public HttpApiHandler(DurableTaskExtension config, TraceWriter traceWriter)
                     output = status.Output,
                     createdTime = status.CreatedTime.ToString("s") + "Z",
                     lastUpdatedTime = status.LastUpdatedTime.ToString("s") + "Z",
-                });
+                    historyEvents = status.History,
+                }) :
+                request.CreateResponse(
+                statusCode,
+                new
+                {
+                    runtimeStatus = status.RuntimeStatus.ToString(),
+                    input = status.Input,
+                    output = status.Output,
+                    createdTime = status.CreatedTime.ToString("s") + "Z",
+                    lastUpdatedTime = status.LastUpdatedTime.ToString("s") + "Z",
+                 });
```

Comment:
The duplication of logic here makes me a bit nervous. I wonder if we should create a dedicated data contract for this response payload and use an attribute to omit the `historyEvents` property if it is null.  Something like:

```csharp
[DataContract]
class StatusResponsePayload
{
    [DataMember(Name = "runtimeStatus")]
    public string RuntimeStatus { get; set; }

    [DataMember(Name = "input")]
    public JToken Input { get; set; }

    [DataMember(Name = "output")]
    public JToken Output { get; set; }

    [DataMember(Name = "createdTime")]
    public string CreatedTime { get; set; }

    [DataMember(Name = "lastUpdatedTime")]
    public string LastUpdatedTime { get; set; }

    [DataMember(Name = "historyEvents", EmitDefaultValue = false)]
    public JArray HistoryEvents { get; set; }
}
```

Depending on how you put together the history events data, this property could be either a `JArray` or a generic array of a new data contract type that you add.

Reply:
@cgillum  This is great suggestion! I was trying to play a bit and modify the anonymous type yesterday but I decided to leave in this unpleasant way just to push the first commit. I was thinking as well for a dedicated data structure but I wasn't sure if it will be OK to introduce it. Now it is clear. Thank you for the code snippet as well!